File: abstractions/ruby/microsoft_kiota_abstractions/lib/microsoft_kiota_abstractions/serialization/serialization_writer.rb
module MicrosoftKiotaAbstractions
  module SerializationWriter

    def write_string_value(key, value)
      raise NotImplementedError.new
    end

    def write_boolean_value(key, value)
      raise NotImplementedError.new
    end

    def write_number_value(key, value)
      raise NotImplementedError.new
    end

    def write_float_value(key, value)
      raise NotImplementedError.new
    end

    def get_date_value(key, value)
      raise NotImplementedError.new
    end

    def write_guid_value(key, value)
      raise NotImplementedError.new
    end

    def write_date_value(key, value)
      raise NotImplementedError.new
    end

    def write_collection_of_primitive_values(key, value)
      raise NotImplementedError.new
    end

    def write_collection_of_object_values(key, value)
      raise NotImplementedError.new
    end

    def write_enum_value(key, value)
      raise NotImplementedError.new
    end

    def get_serialized_content()
      raise NotImplementedError.new
    end

    def write_additional_data(type)
      raise NotImplementedError.new
    end

    def write_any_value(key, value)
      raise NotImplementedError.new
    end
    
  end
end

File: abstractions/swift/Source/MicrosoftKiotaAbstractions/Serialization/SerializationWriter.swift
import Foundation

public protocol SerializationWriter {
    func writeStringValue(key : String, value : String?) throws
    func writeBoolValue(key : String, value : Bool?) throws
    func writeUint8Value(key : String, value : UInt8?) throws
    func writeInt8Value(key : String, value : Int8?) throws
    func writeInt32Value(key : String, value : Int32?) throws
    func writeInt64Value(key : String, value : Int64?) throws
    func writeFloat32Value(key : String, value : Float32?) throws
    func writeFloat64Value(key : String, value : Float64?) throws
    func writeByteArrayValue(key : String, value : [UInt8]) throws
    func writeDateValue(key : String, value : Date?) throws //TODO
    func writeTimeOnlyValue(key : String, value : Date?) throws //TODO
    func writeDateOnlyValue(key : String, value : Date?) throws
    func writeDurationValue(key : String, value : Date?) throws //TODO
    func writeUUIDValue(key : String, value : UUID?) throws
    func writeObjectValue<T: Parsable>(key: String, value: T) throws
    func writeCollectionOfObjectValues<T: Parsable>(key: String, value: [T]) throws
    func writeCollectionOfPrimitiveValues<T>(key: String, value: [T]) throws
    func getSerializedContent() throws -> Data?
    func writeAdditionalData(value: [String:Any]?) throws
    //TODO write enum value & write collection of enum values
}